tests/Models/UserAuthenticationNonceModelTest.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+ * @author Chris Chabilall <chris.c@vanillaforums.com>
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license GPLv2
+ */
+
+namespace VanillaTests\Models;
+
+use VanillaTests\SharedBootstrapTestCase;
+use UserAuthenticationNonceModel;
+use VanillaTests\SiteTestTrait;
+
+/**
+ * Test the {@link UserAuthenticationNonceModel}.
+ */
+class UserAuthenticationNonceModelTest extends SharedBootstrapTestCase {
+    use SiteTestTrait;
+
+    /**
+     * Test nonce is issued and verified
+     */
+    public function testIssueAndVerify() {
+        $model = new UserAuthenticationNonceModel('hhh');
+        $nonce = $model->issue();
+        $this->assertEquals(true,  $model->verify($nonce));
+    }
+
+    /**
+     * That a nonce can be consumed.
+     *
+     * @expectedException \Exception
+     * @expectedExceptionMessage Nonce has expired.
```

Comment:
I think you'll find this fails when you use the proper exception message: Nonce was already used.